jumpy/_custom_fns.py

```diff
@@ -0,0 +1,106 @@
+"""Custom implemented functions."""
+
+from __future__ import annotations
+
+from typing import Any, Callable, Sequence, TypeVar
+
+import numpy as onp
+
+from jumpy import is_jax_installed, ndarray
+from jumpy.core import is_jitted, which_dtype, which_np
+
+try:
+    import jax
+    import jax.numpy as jnp
+except ImportError:
+    jax, jnp = None, None
+
+
+F = TypeVar("F", bound=Callable)
+
+
+def array(object: Any, dtype=None) -> ndarray:
+    """Creates an array given a list."""
+    if dtype is None:
+        try:
+            np = which_np(*object)
+        except TypeError:
+            np = which_np(object)  # object is not iterable (e.g. primitive type)
+    else:
+        np = which_dtype(dtype)
+
+    return np.array(object, dtype)
+
+
+def index_update(x: ndarray, idx: ndarray, y: ndarray) -> ndarray:
+    """Pure equivalent of x[idx] = y."""
+    if which_np(x, idx, y) is jnp:
+        return jnp.array(x).at[idx].set(jnp.array(y))
+    else:
+        x = onp.copy(x)
+        x[idx] = y
+        return x
```

Comment:
Instead of this function, we may want to mimic the jax API for index updates using the `.at[idx].set(y)` syntax (found [here](https://jax.readthedocs.io/en/latest/_autosummary/jax.numpy.ndarray.at.html)). 

I guess we can do that in a separate PR. Would be nice to have that for the `1.0.0` version as well.

We should investigate how we can add that functionality to `numpy.ndarray` in a neat without subclassing.

Reply:
I may find some time this weekend to have a stab at this

Reply:
Ok, that would be great